rules-tests/TypeDeclaration/Rector/Property/PropertyTypeDeclarationRector/Fixture/skip_untyped_inherit_doc.php.inc

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Property\PropertyTypeDeclarationRector\Fixture;
+
+class A {
+    public $property;
+}
+
+final class B {
+    /** @inheritDoc */
+    public $property = 'test';
```

Comment:
Here the `string` type should be completed as `inheritDoc` doc not provide any type with no parent class.

How does PHPStan sees this property type?

Reply:
I've fixed it to have a parent class.

PHPStan reports the types as `mixed` for both of the classes: https://phpstan.org/r/cf7b2a59-f4fc-47b2-b6ef-96ec6c4b4ab0

No type should be completed now (that it extends another class with a property) because it would break variance:
https://phpstan.org/r/3fa1557c-69ed-4ae6-b7cf-2be90e78bb1e

Reply:
In this case it's public, that's why it should be skipped.

> PHPStan reports the types as mixed for both of the classes: phpstan.org/r/cf7b2a59-f4fc-47b2-b6ef-96ec6c4b4ab0

That's why the infer type is added, as PHPStan does not recognize it.

Could you add a separate fixture with inheritdoc, private property and no parent class? In that case it should be decorated with `string` type

Reply:
Done, although I don't think it matters whether it has a parent class or not? This:

```
class A {
	private int $something = 123;
}

class B extends A
{
    private string $something = 'test';
}
```

is 100% valid, so I think it's safe to ignore `@inheritDoc` for all private props?

Reply:
Aside from that, if the type does change I think we should also remove the `@inheritDoc` annotation? Current behaviour is

```diff
 final class A {
-    /** @inheritDoc */
-    private string $property = 'test';
+    /** @inheritDoc
+     * @var string */
+    private $property = 'test';
 }
 ?>
```

and it's almost definitely not correct :)